src/opnsense/www/js/opnsense_bootgrid_plugin.js

```diff
@@ -455,3 +455,32 @@ $.fn.UIBootgrid = function (params) {
         }
     }));
 };
+
+class OPNsenseBasicBootGridTable extends HTMLTableElement {
+
+    /**
+     * constructor - create a new spinner for the button
+     */
+    constructor() {
+        super();
+    }
+
+    /**
+     * connected callback - will be called when the element is injected into the DOM
+     */
+    connectedCallback() {
+
+        const base = this.dataset.base || "";
+        const opts = {};
+        const that = this;
+        ['search', 'get', 'set', 'add', 'del', 'toggle', 'info'].forEach(function (value) {
+            let val = that.dataset[value];
+            if (val) {
+                opts[value] = base + val;
+            }
+        });
+        $(this).UIBootgrid(opts);
+    }
+}
+
+customElements.define('os-bootgrid-table', OPNsenseBasicBootGridTable, {"extends": 'table'});
```

Comment:
browser support might be problematic, although I haven't tested this (https://developer.microsoft.com/en-us/microsoft-edge/platform/status/customelements/)

Reply:
I thought edgeHTML is/was dropped for the chromium engine?

I tested it under Linux with Firefox 70 which also tells that it is experimental but works quite well.
The Mozilla page says they work with edge 76:
https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements

Reply:
The more problematic support would be safari from the Mozilla link.